veniq/dataset_collection/augmentation.py

```diff
@@ -251,42 +251,52 @@ def insert_code_with_new_file_creation(
-    text_lines = read_text_with_autodetected_encoding(str(file_path)).split('\n')
-    line_to_csv = []
-    if body_start_line != body_end_line:
-        algorithm_type = determine_algorithm_insertion_type(
-            ast,
-            method_node,
-            invocation_node,
-            dict_original_invocations
-        )
-        algorithm_for_inlining = AlgorithmFactory().create_obj(algorithm_type)
-        if algorithm_type != InlineTypesAlgorithms.DO_NOTHING:
-            line_to_csv = [
-                file_path,
-                class_name,
-                text_lines[invocation_node.line - 1].lstrip(),
-                invocation_node.line,
-                original_func.line,
-                method_node.name,
-                new_full_filename,
-                body_start_line,
-                body_end_line
-            ]
-
-            algorithm_for_inlining().inline_function(
-                file_path,
-                invocation_node.line,
-                body_start_line,
-                body_end_line,
-                new_full_filename,
+    ncss = NCSSMetric().value(ast.get_subtree(original_func))
+    line_to_csv = {}
+    if ncss > 3:
+        body_start_line, body_end_line = method_body_lines(original_func, file_path)
+        text_lines = read_text_with_autodetected_encoding(str(file_path)).split('\n')
+        if body_start_line != body_end_line:
+            algorithm_type = determine_algorithm_insertion_type(
+                ast,
+                method_node,
+                invocation_node,
+                dict_original_invocations
             )
+            algorithm_for_inlining = AlgorithmFactory().create_obj(algorithm_type)
+            if algorithm_type != InlineTypesAlgorithms.DO_NOTHING:
+                line_to_csv = {
+                    'input_filename': file_path,
+                    'class_name': class_name,
+                    'invocation_text_string': text_lines[invocation_node.line - 1].lstrip(),
+                    'method_where_invocation_occurred': method_node.name,
+                    'start_line_of_function_where_invocation_occurred': method_node.line,
+                    'invocation_method_name': original_func.name,
+                    'invocation_method_start_line': body_start_line,
+                    'invocation_method_end_line': body_end_line,
+                    'output_filename': new_full_filename,
+                }
+
+                algorithm_for_inlining().inline_function(
+                    file_path,
+                    invocation_node.line,
+                    body_start_line,
+                    body_end_line,
+                    new_full_filename,
+                )
+
+                # if get_ast_if_possible(Path(r'D:\temp\AbstractComponent_addBefore_259.java')):
+                if get_ast_if_possible(Path(new_full_filename)):
+                    can_be_parsed = True
+                else:
+                    can_be_parsed = False
```

Comment:
what does it mean that a file cannot be parsed?
do we even want to add such files to the dataset?

Reply:
if we add some code, we can fail the java file. We check it everytime whether a new file is parsed via AST framework.
We check it's correctness

Reply:
yeah, but if we add some code and it fails a java file, we don't want to have such an instance in the dataset, or do we?

Reply:
we will exclude it, we need to know the examples to fix them in future. We can filter it easily